saleor/graphql/menu/mutations.py

```diff
@@ -142,3 +144,52 @@ class Meta:
     @classmethod
     def user_is_allowed(cls, user, input):
         return user.has_perm('menu.manage_menus')
+
+
+class AssignMainNavigation(BaseMutation):
+    menu = graphene.Field(Menu, description='Main navigation bar.')
+
+    class Arguments:
+        menu = graphene.ID(
+            description='ID of the menu to assign as the main navigation bar.')
+
+    class Meta:
+        description = 'Assigns storefront\'s main navigation bar.'
+
+    @classmethod
+    @permission_required(['menu.manage_menus', 'site.manage_settings'])
+    def mutate(cls, root, info, menu=None):
+        errors = []
+        site_settings = info.context.site.settings
+        if menu is not None:
+            menu = cls.get_node_or_error(
+                info, menu, errors=errors, field='menu')
+        if not errors:
+            site_settings.top_menu = menu
+            site_settings.save(update_fields=['top_menu'])
+        return AssignMainNavigation(menu=menu, errors=errors)
+
+
+class AssignSecondaryNavigation(BaseMutation):
```

Comment:
I wonder if we could add an `enum` (or example `PRIMARY/SECONDARY`) flag to the `Arguments`, so we would have only one mutation + remove some boilerplate code. Would it be practical?